Flaky test: test_distributed_respect_user_timeouts/test.py::test_reconnect[configs-ready_to_wait-node1-distributed]

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_distributed_respect_user_timeouts/test.py::test_reconnect[configs-ready_to_wait-node1-distributed]
Failure reason: test_distributed_respect_user_timeouts/test.py:144 - in _check_timeout_and_exception
CI report: [Integration tests (amd_tsan, 2/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=93423&sha=9f1ac837b499cba97a4edc4db5c5ecaae9a627de&name_0=PR&name_1=Integration%20tests%20%28amd_tsan%2C%202%2F6%29)

CIDB statistics: [cidb](https://play.clickhouse.com/play?user=play&run=1#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2Rpc3RyaWJ1dGVkX3Jlc3BlY3RfdXNlcl90aW1lb3V0cy90ZXN0LnB5Ojp0ZXN0X3JlY29ubmVjdFtjb25maWdzLXJlYWR5X3RvX3dhaXQtbm9kZTEtZGlzdHJpYnV0ZWRdJwogICAgLS0gQU5EIGNoZWNrX25hbWUgPSAnSW50ZWdyYXRpb24gdGVzdHMgKGFtZF90c2FuLCAyLzYpJwogICAgQU5EIHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicpCiAgICBBTkQgKChwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCBBTkQgaGVhZF9yZWYgPSAnZml4X3FjY19tZW1vcnlfZGJfZW5naW5lJykgT1IgKHB1bGxfcmVxdWVzdF9udW1iZXIgIT0gMCBBTkQgYmFzZV9yZWYgPSAnbWFzdGVyJykpCkdST1VQIEJZIGRheQpPUkRFUiBCWSBkYXkgREVTQwo=)

Test output:
```
File: test_distributed_respect_user_timeouts/test.py:171 - in test_reconnect
    _check_timeout_and_exception(node, first_user, query_base, query)
File: test_distributed_respect_user_timeouts/test.py:144 - in _check_timeout_and_exception
    assert (
E   assert (18.04602519900004 - 15) <= 3
```

--- a/tests/integration/test_distributed_respect_user_timeouts/test.py
+++ b/tests/integration/test_distributed_respect_user_timeouts/test.py
@@ -11,8 +11,6 @@ from helpers.test_tools import TSV
 
 NODES = {"node" + str(i): None for i in (1, 2)}
 
-IS_DEBUG = False
-
 CREATE_TABLES_SQL = """
 CREATE DATABASE test;
 
@@ -111,13 +109,6 @@ def started_cluster(request):
     try:
         cluster.start()
 
-        if cluster.instances["node1"].is_debug_build():
-            global IS_DEBUG
-            IS_DEBUG = True
-            logging.warning(
-                "Debug build is too slow to show difference in timings. We disable checks."
-            )
-
         for node_id, node in list(NODES.items()):
             node.query(CREATE_TABLES_SQL)
             node.query(INSERT_SQL_TEMPLATE.format(node_id=node_id))
@@ -139,12 +130,11 @@ def _check_timeout_and_exception(node, user, query_base, query):
     # And it should timeout no faster than:
     measured_timeout = timeit.default_timer() - start
 
-    if not IS_DEBUG:
-        assert expected_timeout - measured_timeout <= TIMEOUT_MEASUREMENT_EPS
-        assert (
-            measured_timeout - expected_timeout
-            <= TIMEOUT_DIFF_UPPER_BOUND[user][query_base]
-        )
+    assert expected_timeout - measured_timeout <= TIMEOUT_MEASUREMENT_EPS
+    assert (
+        measured_timeout - expected_timeout
+        <= TIMEOUT_DIFF_UPPER_BOUND[user][query_base]
+    )
 
     # And exception should reflect connection attempts:
     _check_exception(exception, repeats)
@@ -156,6 +146,9 @@ def _check_timeout_and_exception(node, user, query_base, query):
 )
 def test_reconnect(started_cluster, node_name, first_user, query_base):
     node = NODES[node_name]
+    if node.is_built_with_sanitizer() or node.is_debug_build():
+        pytest.skip("Disabled for debug and sanitizer builds")
+
     query = SELECTS_SQL[query_base]
     if started_cluster.__with_ssl_config:
         query = query.replace("remote(", "remoteSecure(")

